CONTRIBUTING.md

```diff
@@ -110,6 +110,12 @@ splitting it up into many small single-use functions. Single-caller functions
 can be useful, but consider whether the code would be easier to follow if the
 function was inlined (especially for short functions).
 
+#### Consider avoiding free-standing functions
+
+Often free-standing functions take arguments that could be associated with
```

Comment:
Maybe elaborate on this a bit? For example, if a function's semantics or implementation are strongly dependent on one of its arguments and that argument is defined in the current crate, it should probably become a method. (Also consider the case where a function is getting multiple arguments that originate from the same common type in all callers.)

Reply:
I borrowed some of your wording for this. Thanks.